### PR TITLE
Atom/Destroy now operates on contents

### LINKED ISSUE #18346: Blackscreen when "dying" as a possessed object
1) Possess a supermatter shard.
2) Make a huge explosion that destroys the shard.
3) You are now dead, in deadchat with the name "supermatter shard", but you cannot see, even if you teleport or orbit other things. A client reconnect solves this.

Possibly related to #18316 ?

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #18346

Give me a good reason not to...

### CODE REVIEW

**Comment by coiax** [comment]
This is going to expose/cause a number of odd bugs. But it's probably for the best.

**Comment by xxalpha** [comment]
I thought it already did, in order to qdel correctly?

**Comment by Cyberboss** [comment]
Loc isn't a variable that prevents gcing I BELIEVE

**Comment by RemieRichards** [comment]
your beliefs are odd, ```loc``` is the PRIMARY EXAMPLE of doing garbage collection in DM.
very basic "gc" in DM is usually just a proc that does ```loc = null```, it's literally a requirement to GC.

**Comment by Cyberboss** [comment]
I meant in reverse order. Eg. If a closet is qdeld and didn't dump it's
shit, the loc vars of the contents won't prevent it from being gc'd. Unless
I am mistaken

**Comment by RemieRichards** [comment]
in that respect I believe loc is just a normal reference, thus preventing GC.
typically we deleted things inside other things in the container's Destroy() anyway, which is why it went largely unnoticed.

**Comment by optimumtact** [comment]
The reason is you can't just change something so fundamental and expect everything to respect it and work properly, there's no way to tell if there is anything incorrectly relying on this behaviour for reasons that are not clear.

**Comment by coiax** [comment]
This definitely would need test merging, because I bet a whole bunch of weird bugs would immediately crop up.

File: code/game/atoms.dm [hunk 1]
```diff
@@ -93,6 +93,10 @@
 	LAZYCLEARLIST(priority_overlays)
 	//SSoverlays.processing -= src	//we COULD do this, but it's better to just let it fall out of the processing queue
 
+	//anything not dead by this point needs to fucking not exist
+	for(var/I in src)
+		qdel(I)
```
Comment (duncathan) [inline code/game/atoms.dm]: after this loop is finished you need to do `contents.Cut()` so that the references to these atoms get removed

**Review by duncathan — CHANGES_REQUESTED** [review]

File: code/game/atoms.dm (hunk 1 above)
Comment (xxalpha) [inline code/game/atoms.dm]: Once the object is deleted those references go away.

Comment (Cyberboss) [inline code/game/atoms.dm]: Yeah, atom/movable/Destroy sets `loc` to null, so the list will be cleared by itself

Comment (duncathan) [inline code/game/atoms.dm]: oh ya duh

Comment (duncathan) [inline code/game/atoms.dm]: I forgot about the automatic connection between loc and contents

**Comment by duncathan** [comment]
I'm gonna merge this - any issues that arise will only be symptoms of existing issues being uncovered and can be fixed on a case by case basis

**Comment by Cyberboss** [comment]
O.o

**Comment by optimumtact** [comment]
@duncathan what were you thinking? this should have been test merged at least once before being merged

/me slaps wrist

**Pull request merged**